test/utils/cluster.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	"k8s.io/klog/v2"
+)
+
+func EnsureCluster(location, clusterName string, numOfNodes int) error {
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", location,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
+	if err != nil {
```

Comment:
How do we know that this error is a NotFound error and not something else?

It makes more sense to list all of the clusters in the zone and if the cluster doesn't show up in the list, then you know that the cluster needs to be created. 

Otherwise, bubble up the error.

Reply:
Updated.